### PR TITLE
Update download step in east sys-setup command

### PR DESCRIPTION
**Pull request by MarkoSagadin**
# Description

This PR adds new packages that will be installed by the east sys-setup command.
Download function was also updated in this process.

Related: #86

## Areas of interest

Added code in `sys_setup.py`.

## Checklist

- [x] Source code documentation for all newly added or changed functions was added/updated.

## After-review steps

* Code author will merge the PR by himself/herself.

### CODE REVIEW

**Comment by MarkoSagadin** [comment]
Current dependencies on/for this PR:
* dev
  * **PR #87** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-east-software/87" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/IRNAS/irnas-east-software/87?utm_source=stack-comment).

File: src/east/system_commands/sys_setup.py [hunk 1]
```diff
@@ -82,6 +83,43 @@ def _get_toolchain_download_link():
...
 
+def _get_cppcheck_download_link():
+    """Just a convenience function for getting the link for the Cppcheck source code.
+    This link will be updated regularly.
+
+    Note: If you change version do not forget to update the folder name in the
+    _install_cppcheck below!
+    """
+
+    link = "https://github.com/danmar/cppcheck/archive/refs/tags/2.12.0.tar.gz"
```
Comment (TjazVracko) [inline src/east/system_commands/sys_setup.py:101]: You could make the link, or at least the version a constant in the `constans.py` file. Then, this function and `_install_cppcheck` would use the same "source of truth".
You could then remove the Note and will also never forget to update `_install_cppcheck`

File: src/east/system_commands/sys_setup.py [hunk 2]
```diff
@@ -120,6 +122,54 @@ def _install_cppcheck(east: EastContext, exe_path: str):
...
+    return link
+
+
+def _install_clang_llvm(east: EastContext, exe_path: str):
+    """Installs clang+llvm to a proper location"""
+
+    clang_dir = os.path.join(east.consts["tooling_dir"], "clang+llvm")
+
+    # Remove old cppcheck dir if it exists
+    shutil.rmtree(clang_dir, ignore_errors=True)
```
Comment (TjazVracko) [inline src/east/system_commands/sys_setup.py:167]: Did you mean _"remove old clang_dir"_ ?

File: src/east/system_commands/sys_setup.py [hunk 3]
```diff
@@ -120,6 +122,54 @@ def _install_cppcheck(east: EastContext, exe_path: str):
...
+
+    east.print("[bold blue]Extracing clang+llvm binaries...")
+    shutil.unpack_archive(exe_path, east.consts["tooling_dir"], format="gztar")
+    east.print("[bold green]Done!")
+
+    # Remove version from the folder name
+    dir = glob.glob(os.path.join(east.consts["tooling_dir"], "clang+llvm-*"))[0]
+    shutil.move(dir, clang_dir)
```
Comment (TjazVracko) [inline src/east/system_commands/sys_setup.py:175]: You use a folder name without a version for clang+llvm and a folder with a version for cppcheck. Is this intended? 

Are you meant to install multiple versions of cppcheck and only one version of clang?

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ### Added
 
--   east sys-setup now also installs cppcheck and clang+llvm.
+-   east sys-setup now also installs cppcheck, clang+llvm and CodeChecker.
```
Comment (TjazVracko) [inline CHANGELOG.md:11]: You could have completed 3x more features by rewording this:

```
-   east sys-setup now also installs cppcheck.
-   east sys-setup now also installs clang+llvm.
-   east sys-setup now also installs CodeChecker.
```

:p

**Review by TjazVracko — COMMENTED** [review]
Excellent stuff.
I added some thoughts, but there is nothing mayor to change.

File: src/east/system_commands/sys_setup.py (hunk 2 above)
Comment (MarkoSagadin) [reply src/east/system_commands/sys_setup.py:168]: You are right

File: src/east/system_commands/sys_setup.py (hunk 3 above)
Comment (MarkoSagadin) [reply src/east/system_commands/sys_setup.py:176]: No, the issue is that the downloaded cppcheck tarball extracts into a folder with the name format `cppcheck-<version>`.

Line 170, performs rename.

File: src/east/system_commands/sys_setup.py (hunk 1 above)
Comment (MarkoSagadin) [reply src/east/system_commands/sys_setup.py:102]: I removed the explicit cppcheck version from the `__install_cppcheck` function and replaced it with glob function.

The purpose of constants.py is a bit different, I only store there various paths that should not ever change, but should be available to the rest of the East.

The version links are used only in sys_setup.py file.

**Review by TjazVracko — APPROVED** [review]

**Review by TjazVracko — APPROVED** [review]

**Pull request merged**